/*
 * Copyright 2019 The Feast Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

import "feast/types/Field.proto";

package feast.storage;

option java_outer_classname = "RedisProto";
option java_package = "feast.proto.storage";
option go_package = "github.com/feast-dev/feast/sdk/go/protos/feast/storage";

message RedisKey {
  // Field number 1 is reserved for a future distributing hash if needed
  // (for when redis is clustered).

  // FeatureSet this row belongs to, this is defined as featureSetName.
  string feature_set = 2;

  // List of fields containing entity names and their respective values
  // contained within this feature row. The entities should be sorted 
  // by the entity name alphabetically in ascending order.
  repeated feast.types.Field entities = 3;
}
